Txr/EncVelManager.cpp

```diff
@@ -51,17 +51,15 @@ void EncVelManager::SetLEDs(char force)
     ENC_GREEN_LED_OFF();
     ENC_RED_LED_OFF();
   }
-  // if we've crossed the midway point make sure lights are all the way off
-  else if (mPrevVelocityPercent < MID_VELOCITY ^ mVelocityPercent < MID_VELOCITY) {
-    SetAllLEDsOff();
-  }
   
   if (mVelocityPercent == MID_VELOCITY) {
+    SetAllLEDsOff();
     ENC_GREEN_LED_ON();
     ENC_RED_LED_ON();
   }
   else if (mVelocityPercent > MID_VELOCITY) {
     int dim = map(mVelocityPercent, MID_VELOCITY + 1, MAX_VELOCITY, 1, 250);
+    analogWrite(SPEED_LED2, 0);
```

Comment:
Hmm don't really remember, that was 2 months ago and now I don't have my LH at hand to try again... But I know that I could get a LED to stay stuck consistently every time I'd rotate the encoder quickly, making it quite unlikely it would be due to sampling right on MID_VELOCITY. That made me suspect the XOR, backed by the fact that the XOR operator in C is a bitwise operator and not a logical one and that can have strange effects on variables that are not explicitly declared as booleans. 
Decided to get rid of it for clarity and safety. 
That statement could probably get rewritten to work, but then I'm not sure if evaluating the nested conditions would really execute faster than just defining the output to a known state like I did here, and if it did if it would be of any significance.